File: public/examples/ruby/basic/04_control_flow.rb
# Ruby Basic: Control Flow

# If / Unless
rainy = false
puts "Go outside" unless rainy

# Case statement
grade = "A"
case grade
when "A"
  puts "Excellent"
when "B"
  puts "Good"
else
  puts "Try harder"
end

# Loop
3.times { |i| puts "Iteration #{i}" }

File: examples/zig/basic/04_control_flow.zig
const std = @import("std");

pub fn main() !void {
    const score = 85;

    // If-Else
    if (score >= 90) {
        std.debug.print("Grade: A\n", .{});
    } else if (score >= 80) {
        std.debug.print("Grade: B\n", .{});
    } else {
        std.debug.print("Grade: C\n", .{});
    }

    // While loop
    var i: u32 = 0;
    while (i < 3) : (i += 1) {
        std.debug.print("i: {d}\n", .{i});
    }
}
